arraystask/script.js

```diff
@@ -0,0 +1,101 @@
+function isArray2(a) {
+  return a instanceof Array;
+}
+
+function isArray3(a) {
+  return Object.prototype.toString.call(a) == '[object Array]';
+}
+
+function range(a, b, step) {
+  var result = [];
+  if (!b) {
+    b = a;
+    a = 0;
+  }
+  if (!step) {
+    step = 1;
+  }
+  var n = Math.ceil((b - a) / step);
+  for (var i = 0; i < n ; i++) {
+    result.push(a);
+    a += step;
+  }
+  return result;
+}
+
+function compact1(a) {
+  return a.filter(Boolean);
+}
+
+function compact2(a) {
+  var truthy = [];
+  for (var i = 0; i < a.length; i++) {
+    if (a[i]) {
+      truthy.push(a[i]);
+    }
+  }
+  return truthy;
+}
+
+function sum1(a) {
+  var add = function(x, y) {
+    return x + y;
+  };
+  return a.reduce(add, 0);
+}
+
+function sum2(a) {
+  var res = 0;
+  for (var i = 0; i < a.length; i++) {
+    res += a[i];
+  }
+  return res;
+}
+
+function unique(a) {
+  var filterCb = function(value, index, self) {
+    return self.indexOf(value) === index;
+  }
+  return a.filter(filterCb);
+}
+
+function last(a) {
+  return a[a.length - 1];
+}
+
+function excludeLast(a, b) {
+  if ("undefined" == typeof b) {
+    b = 1;
+  }
+  return a.slice(0, a.length - b);
```

Comment:
one disadvantage of this approach is when b is negative slice will return old array + [undef, undef, undef]. you can try it